src/CommitmentsV2/SFA.DAS.CommitmentsV2/Services/TransferRequestDomainService.cs

```diff
@@ -0,0 +1,118 @@
+
+        public TransferRequestDomainService(Lazy<ProviderCommitmentsDbContext> dbContext,
+            ILogger<TransferRequestDomainService> logger)
+        {
+            _dbContext = dbContext;
+            _logger = logger;
+        }
+
+        public async Task ApproveTransferRequest(long transferRequestId, UserInfo userInfo, DateTime approvedOn, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var transferRequest = await _dbContext.Value.TransferRequests
+                    .Include(c => c.Cohort)
+                    .Where(x => x.Id == transferRequestId)
+                    .SingleAsync(cancellationToken);
+
+                if (transferRequest.Status == TransferApprovalStatus.Approved)
+                {
+                    _logger.LogWarning($"Transfer Request {transferRequest.Id} has already been approved");
+                    return;
+                }
+
+                transferRequest.Approve(userInfo, approvedOn);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Error processing {nameof(ApproveTransferRequest)}");
+                throw;
+            }
+        }
+
+        public async Task RejectTransferRequest(long transferRequestId, UserInfo userInfo, DateTime rejectedOn, CancellationToken cancellationToken)
+        {
+            try
+            {
+                var transferRequest = await _dbContext.Value.TransferRequests
+                    .Include(c => c.Cohort)
+                    .Where(x => x.Id == transferRequestId)
+                    .SingleAsync();
+
+                if (transferRequest.Status == TransferApprovalStatus.Rejected)
+                {
+                    _logger.LogWarning($"Transfer Request {transferRequestId} has already been rejected");
+                    return;
+                }
+
+                transferRequest.Reject(userInfo, rejectedOn);
+            }
+            catch (Exception ex)
+            {
+                _logger.LogError(ex, $"Error processing {nameof(RejectTransferRequest)}");
+                throw;
+            }
+        }
+
+        public async Task<GetTransferRequestQueryResult> GetTransferRequest(long transferRequestId, long employerAccountId)
+        {
+            _logger.LogInformation($"Getting Transfer Request {transferRequestId} for employer account {employerAccountId}");
+
+            var query = from t in _dbContext.Value.TransferRequests
+                        join c in _dbContext.Value.Cohorts
+                            on t.CommitmentId equals c.Id
+                        join a in _dbContext.Value.Accounts
+                            on c.TransferSenderId equals a.Id
+                        join ale in _dbContext.Value.AccountLegalEntities
+                            on c.AccountLegalEntityId equals ale.Id
+                        where
+                            t.Id == transferRequestId
```

Comment:
for these straightforward joins, would .Include() for each entity not have the same effect?

Reply:
Done